Geometry_Engine/Modify/Offset.cs

```diff
@@ -0,0 +1,868 @@
+                        j = 0;
+                    temp = offsetCurves[i].Fillet(offsetCurves[j], tangentExtensions, true, false, tolerance);
+                }
+                if (!(temp == null)) //inserting filetted curves
+                {
+                    if (j != 0)
+                    {
+                        offsetCurves.RemoveRange(i, 2);
+                        offsetCurves.InsertRange(i, temp.Curves);
+                    }
+                    else
+                    {
+                        offsetCurves.RemoveAt(i);
+                        offsetCurves.RemoveAt(0);
+                        offsetCurves.InsertRange(i - 1, temp.Curves);
+                    }
+                    i = i + temp.Curves.Count - 2;
+                }
+                else
+                    connectingError = true;
+            }
+
+            //removing curves that are to close to the main curve
+            for (int i = 0; i < offsetCurves.Count; i++)
+            {
+                if ((offsetCurves[i].IPointAtParameter(0.5).Distance(curve) + tolerance < Math.Abs(offset) &&
+                     (offsetCurves[i].IStartPoint().Distance(curve) + tolerance < Math.Abs(offset) ||
+                      offsetCurves[i].IEndPoint().Distance(curve) + tolerance < Math.Abs(offset))))
+                {
+                    PolyCurve temp = offsetCurves[((i - 1) + offsetCurves.Count) % offsetCurves.Count].Fillet(offsetCurves[(i + 1) % offsetCurves.Count], tangentExtensions, true, false, tolerance);
+                    if (temp != null)
+                        if (i == 0)
+                        {
+                            offsetCurves.RemoveRange(0, 2);
+                            offsetCurves.RemoveAt(offsetCurves.Count - 1);
+                            offsetCurves.InsertRange(0, temp.Curves);
+                            i = temp.Curves.Count - 1;
+                        }
+                        else if (i == offsetCurves.Count - 1)
+                        {
+                            offsetCurves.RemoveRange(i - 1, 2);
+                            offsetCurves.RemoveAt(0);
+                            offsetCurves.InsertRange(offsetCurves.Count - 1, temp.Curves);
+                            i = offsetCurves.Count - 1;
+                        }
+                        else
+                        {
+                            offsetCurves.RemoveRange(i - 1, 3);
+                            offsetCurves.InsertRange(i - 1, temp.Curves);
+                            i = i - 3 + temp.Curves.Count;
+                        }
+                    if (offsetCurves.Count < 1)
+                    {
+                        log.Clear();
+                        Reflection.Compute.RecordError("Method failed to produce corretct offset. Returning null.");
+                        return null;
+                    }
+                    counter++;
+                }
+            }
+
+            log.Clear();
```

Comment:
Will `log.Clear()` not squash all warnings/errors recoreded up to date?

Reply:
as log is pointing to the `CurrentEvents` it would clear that log up to this point. This is generally cleared before each component run, so will not invalidate all warnings/errors across other components.
Would consider changing this line to:
`Reflection.Compute.ClearCurrentEvents()` 
instead if that is what we want to happen here and if so put some clear comments around why in the code and a TODO to get it removed.
Why is this needed here? No way to get rid of it?

Reply:
@IsakNaslundBh Yes, `ClearCurrentEvents` looks better and seems to be doing the same. Somehow I didn't notice this method. Can replace it.
I described it in #1306. The point is that I use other methods that produce warning/errors which I am aware of and they don't have an impact on final solution. I want to clear them so they won't be visible while using Offset component.

Reply:
Use `ClearCurrentEvents` and add a TODO if valid :smile: